### PR TITLE
Complete deprecation work (#429)

### PR DESCRIPTION
**Pull request by bartlettroscoe**
### Description

There was a good bit of work left over from #429 that I noticed after I updated the snapshot of TriBITS in Trilinos and saw a bunch of deprecation warnings.  I also saw that TriBITS examples were using deprecated code as well.

The main commit in this PR is https://github.com/TriBITSPub/TriBITS/commit/69fb01c520c454fb2fe049aadf21263a75b2358c which updates the documentation and fixes some bugs.

This PR also does some other things (see the individual commits for details).

### Instructions for reviewers

Please review the commits one-by-one.  That will make the review the easiest to do and provide the best feedback.

### CODE REVIEW

**Comment by bartlettroscoe** [comment]
Hello @KyleFromKitware, can you please give this PR a quick review?  I would like to get this PR merged as part of the TriBITS update of Trilinos in https://github.com/trilinos/Trilinos/pull/11380.

File: refactoring/token-replace.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+
+def getCmndLineOptions():
+  from argparse import ArgumentParser, RawDescriptionHelpFormatter
+
+  clp = ArgumentParser(description=usageHelp,
+    formatter_class=RawDescriptionHelpFormatter)
+
+  clp.add_argument(
+    "-t", dest="tokenToReplace", required=True,
```
Comment (KyleFromKitware) [inline refactoring/token-replace.py:21]: May want to validate that this argument is a valid token (`^[a-zA-Z0-9_]+$`).

File: refactoring/token-replace.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+#
+
+if __name__ == '__main__':
+
+  import sys, re
+
+  inOptions = getCmndLineOptions()
+
+  beginLineTokenPattern = re.compile(
+    r'^' + inOptions.tokenToReplace + r'([^A-Za-z0-9_])' )
```
Comment (KyleFromKitware) [inline refactoring/token-replace.py:55]: Using `f''` syntax may make this more readable.

File: refactoring/token-replace.py (hunk 1 above)
Comment (bartlettroscoe) [reply refactoring/token-replace.py:21]: If I add unit tests for these simple tools, I will keep that in mind.  One would always carefully examine the impact of running this tool to replace tokens by running `git diff` to make sure it did what you expect.

File: refactoring/token-replace.py (hunk 2 above)
Comment (bartlettroscoe) [reply refactoring/token-replace.py:55]: Is it possible to mix `r'...'` and `f'...'` in the same sting?

Comment (bartlettroscoe) [reply refactoring/token-replace.py:55]: FYI: I did a direct push to 'master' for this fix in commit https://github.com/TriBITSPub/TriBITS/commit/28a5c830dd51c488ca7ee299ff5fa6239f814c39.

There is no automated testing for that code so there is no point in creating a PR and running all of those GHA builds.

**Pull request merged**